addrs/provider_type.go

```diff
@@ -3,5 +3,21 @@ package addrs
 // ProviderType encapsulates a single provider type. In the future this will be
 // extended to include additional fields including Namespace and SourceHost
 type ProviderType struct {
-	Type string
+	Type      string
```

Comment:
I see your argument against Type stutter, and I'm amenable to changing this. My concern is that `name` is already overloaded and in the future it will be possible for a provider to have a type, such as "terraform", but also a (module-specific) user-supplied _name_.  I want it to be very clear that here I am talking about the "type name", and not necessarily the (new) "name". 

I _can_ use `name` here and (eventually) work on a clear and specific nomenclature for the new name-thing, if that sounds better? I just don't like how unclear providers already are between type, name, and typeName.

Reply:
In chat we talked about renaming this type to `addrs.Provider` instead, at which point it wouldn't stutter anymore. I'm happy with taking that path instead, and then we can plan for future work to make the rest of Terraform consistent that the unqualified noun "provider" means a plugin you can install, and to use qualifiers for all other places we currently use the noun "provider" unqualified.